### ISSUE TITLE
Enhance Audio Offload documentation

### ISSUE THREAD

**Issue by yschimke** [androidx/media#136]
### Use case description

Reviewing 
- https://exoplayer.dev/battery-consumption.html
- https://exoplayer.dev/doc/reference/com/google/android/exoplayer2/ExoPlayer.html#experimentalSetOffloadSchedulingEnabled(boolean)

Topics to consider adding

- limitations of offload - volume normalisation, gapless, device/heaphone compatibility
- difference between sleepingForOffload and audioTrack.isOffloadedPlayback
- detecting offload disabled via onAudioSinkError
- API requirements, 29+

### Proposed solution

Document the tradeoffs for audio offload and the ways to verify status.

### Alternatives considered

None

**Comment by lemondoglol** [comment]
Could we get some update on this thread? There are not much docs we can find online related to AudioOffload even it is out for quite sometime. Appreciate if we can more info! Thanks!

**Comment by tonihei** [comment]
There is some documentation on [how to turn it on on the track selection page](https://developer.android.com/media/media3/exoplayer/track-selection#audioOffload) and it's also [mentioned on the battery consumption page](https://developer.android.com/media/media3/exoplayer/battery-consumption#audio-playback).

@microkatz Do you think it's worth adding a few more sentences around the general restrictions and event listening infrastructure somewhere?

**Comment by yschimke** [comment]
Am I right that since 1.2.0, it's much simplified?

```
                trackSelectionParameters = trackSelectionParameters.buildUpon()
                    .setAudioOffloadPreferences(
                        AudioOffloadPreferences.Builder()
                            .setAudioOffloadMode(AudioOffloadPreferences.AUDIO_OFFLOAD_MODE_ENABLED)
                            .setIsSpeedChangeSupportRequired(false)
                            .setIsGaplessSupportRequired(false)
                            .build(),
                    )
                    .build()
```

This should generally get ExoPlayer to do the right thing for Audio only tracks?

And that watching the process lifecycle and toggling enableSleepingForOffload as the app moves between foreground and background is no longer needed?

**Comment by tonihei** [comment]
@yschimke  Yes, see the docs I linked above. There is nothing else needed to set this up, but it may still be worth documenting the reasons for why you want to do this and what kind of restrictions it may have.

**Comment by lemondoglol** [comment]
in this medium post [ExoPlayer 2.12 - What’s new
](https://medium.com/google-exoplayer/exoplayer-2-12-whats-new-e43ef8ff72e7#:~:text=Speed%20adjustment%20and%20other%20audio%20processing%20effects%20cannot%20be%20used%20in%20conjunction%20with%20audio%20offload%2C%20but%20it%20may%20be%20possible%20to%20use%20some%20framework%20AudioEffects%20with%20offload%20enabled.)

it mentioned speed adjustment, and suggested to use [AudioEffect] to make it work with audio offload. Could we get some info for how to do it as well or any documents for how to do it? Appreciate for it! The feature is nice, but it also has a great cons for it. Would like to better understand this before using it. Thanks again!

**Comment by ronak2121** [comment]
Pinging on this ticket for an update
